eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java

```diff
@@ -110,6 +106,94 @@ void assemblesTogether(@TempDir final Path temp) throws IOException {
+                "      \"Hello five\"",
+                "    QQ.io.stdout|0.28.6",
+                "      \"Hello six\""
+            )
+            .with("withVersions", true)
+            .with(
+                "objectionaries",
+                new ObjsDefault(
+                    new MapEntry<>(master, new OyRemote(master)),
+                    new MapEntry<>(five, new OyRemote(five)),
+                    new MapEntry<>(six, new OyRemote(six))
+                )
+            )
+            .execute(AssembleMojo.class)
+            .result();
+        final String stdout = "**/io/stdout";
+        final String fifth = new OnVersioned(stdout, "17f8929.xmir").toString();
+        final String sixth = new OnVersioned(stdout, "9c93528.xmir").toString();
+        final String path = "target/%s/org/eolang";
+        final String parse = String.format(path, ParseMojo.DIR);
+        final String optimize = String.format(path, OptimizeMojo.DIR);
+        final String pull = String.format(path, PullMojo.DIR);
+        final String resolve = String.format("target/%s", ResolveMojo.DIR);
+        final String seq = "**/seq.xmir";
+        final String string = "**/string.xmir";
+        final String hash = String.join(".", master.value(), "eo");
+        final String[] jars = new String[] {
+            "**/eo-runtime-0.28.5.jar",
+            "**/eo-runtime-0.28.6.jar",
+            "**/eo-runtime-0.30.0.jar",
+        };
+        MatcherAssert.assertThat(
+            String.format(
+                "AssembleMojo should placed parsed files under %s directory, but didn't",
+                parse
+            ),
+            result.get(parse).toAbsolutePath(),
+            new ContainsFiles(fifth, sixth, seq, string)
+        );
+        MatcherAssert.assertThat(
+            String.format(
+                "AssembleMojo should placed optimized files under %s directory, but didn't",
+                optimize
+            ),
+            result.get(optimize).toAbsolutePath(),
+            new ContainsFiles(fifth, sixth, seq, string)
+        );
+        MatcherAssert.assertThat(
+            String.format(
+                "AssembleMojo should placed pulled files under %s directory, but didn't",
+                pull
+            ),
+            result.get(pull).toAbsolutePath(),
+            new ContainsFiles(
+                new OnVersioned(stdout, "17f8929.eo").toString(),
+                new OnVersioned(stdout, "9c93528.eo").toString(),
+                new OnVersioned("**/seq", hash).toString(),
+                new OnVersioned("**/string", hash).toString()
+            )
+        );
+        MatcherAssert.assertThat(
+            String.format(
+                "AssembleMojo should placed pulled files under %s directory, but didn't",
+                resolve
+            ),
+            result.get(resolve).toAbsolutePath(),
+            new ContainsFiles(jars)
+        );
+        MatcherAssert.assertThat(
```

Comment:
@volodya-lombrozo I think we definitely can't test all this in particular mojos, because `AssembleMojo` makes several passes through the cycle. We can split the test into smaller ones, but it affects performance. Here `AssembleMojo` makes 3 passes through its cycle which takes about 4 seconds on my mac (I don't think it would faster in our CI). So if we split the test into five ones (check parsed files, optimized, pulled, resolved, classes) - it will take five times longer.

Reply:
@maxonfjvipon I mean the logic of `AssembleMojo` is to run several phases one-by-one in a loop. Nothing more. 
Everything else you are trying to check here is related to particular mojos (and we should check their behaviour in appropriate tests). In other words you wrote an integration test here. And, probably, it is not the best place for it.